src/server/pipeline/match_data.py

```diff
@@ -27,13 +29,12 @@ def start(connection, added_or_updated_rows, manual_matches_df, job_id):
 
     current_app.logger.info("***** Running execute job ID " + job_id + " *****")
     items_to_update = pd.concat([added_or_updated_rows["new"], added_or_updated_rows["updated"]], ignore_index=True)
-    pdp_contacts = pd.read_sql_table('pdp_contacts', connection)
-
-    if pdp_contacts["matching_id"].dropna().size == 0:
-        max_matching_group = 0
-    else:
-        max_matching_group = max(pdp_contacts["matching_id"].dropna())
 
+    query = "select max(matching_id) as matching_id from pdp_contacts where archived_date is null"
+    max_matching_id = connection.execute(query).fetchone()[0]
```

Comment:
Will you always have a row returned from this? If not, you'll get an error if you try to subscript None.
You may want to do something like
```
result = connection.execute(query) 
if result.rowcount:
   max_matching_id = result.fetchone()[0]
```

Reply:
The aggregate function will always return one result. When the table is empty it returns one result with a null value.